code/script.js

```diff
@@ -0,0 +1,107 @@
+const projects = document.getElementById('projects')
+const profile = document.getElementById('profile')
+const repos = document.getElementById('repos')
+
+const username = 'StephannieMedenilla'
+const USER_PROFILE = `https://api.github.com/users/SteppbySteph`
+const USER_REPO = `https://api.github.com/users/SteppbySteph/repos`
+
+
+
+const options = {
+    method: 'GET',
+    headers: {
+          Authorization: `token ${GITHUB_TOKEN}`
+    }
+}    
+
+
+//----USER NAME & PROFILE----//
+
+const getProfile = () => {
+    fetch(USER_PROFILE, options)
+        .then(res => res.json())
+        .then(data => {
+            const user = data.name
+            const avatar = data.avatar_url
+            profile.innerHTML += `
+            <div class="profile-info">  
+                <img class="avatar" src="${avatar}">
+                <p>${user}</p>
+            </div>
+             `
+         }) 
+}
+
+getProfile()
+
+//----REPOSITORIES WITH DATA OF NAME, PUSH DATE, DEFAULT BRANCH, URL, PULL REQUESTS AND getCommitNr ()----//
+
+const getRepos = () => {
+    fetch(USER_REPO, options)
+        .then(res => res.json())
+        .then(data => {
+            // Creating constsant for my filtered repos
+            forkedRepos = data.filter((repo) => repo.name.startsWith('project-'))
+
+            // Creating variable to pass on to showChart().
+            allMyRepos = forkedRepos.length
+            
+            // forEach function to create HTML elements & pull requests
+            forkedRepos.forEach((repo) => {
+                      
+                // Getting Repo Name
+                repoName = repo.name;
+
+                // Getting Most Recent Push Date which will be formated according to 'substr' method.
+                pushDate = repo.pushed_at.substr(0, 10)
+
+                // Getting the Default Branch                 
+                defaultBranch = repo.default_branch
+
+                // Getting the URL of the repo
+                urlRepo = repo.html_url
+
+                // Creating innerHTML to insert above info
```

Comment:
Really nice commenting as this is difficult parts to get hold of if not explained.